Add Release Notes 1.12 Santiago (#556)
kyma-project/website: content/blog-posts/2020-04-30-release-notes-1.12/index.md

---
title: "Kyma 1.12 Santiago"
author:
  name: "Barbara Czyż, Technical Writer @Kyma"
tags:
  - release-notes
type: release
releaseTag: "1.12.0"
redirectFrom:
  - "/blog/release-notes-112"
---

Even though the current situation calls for some stay-at-home time, we don't lose our sense of adventure. The Kyma ship leaves Rome and embarks on a long, exciting journey all the way to Santiago de Chile, using the time well to deliver a set of new features. Join us in celebration, and once you get a glimpse of the majestic Andes surrounding the city, explore what Kyma 1.12 has to offer. This release brings API Gateway v2 as the sole mechanism for exposing services, introduces refurbished Serverless and a fresh set of features brought in by Compass - all that topped off with brand new Kiali and Tracing. For more details, read the full story behind Kyma 1.12 Santiago.
<!-- overview -->

> > **CAUTION:** Before upgrading your Kyma deployment to 1.12, you must perform the migration steps described in the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.12/docs/migration-guides/1.11-1.12.md). If you upgrade to the new release without performing these steps, you can compromise the functionality of your cluster or make it unusable.

Reviewer: ```suggestion
> **CAUTION:** Before upgrading your Kyma deployment to 1.12, you must perform the migration steps described in the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.12/docs/migration-guides/1.11-1.12.md). If you upgrade to the new release without performing these steps, you can compromise the functionality of your cluster or make it unusable. However, the Migration Guide also contains steps you must complete to migrate your Kubeless Functions to the new solution - complete them after you upgrade to the new release.
```

Reviewer: I added this part: ` However, the Migration Guide also contains steps you must complete to migrate your Kubeless Functions to the new solution - complete them after you upgrade to the new release.`

### Documentation updates

With Event Bus removed, we focused on Knative Eventing Mesh. For 1.12, this also meant providing you with additional documentation on [event processing and delivery](https://kyma-project.io/docs/1.12/components/knative-eventing-mesh/#details-event-processing-and-delivery). If you are interested in setting Kafka as your default Channel implementation, read this [tutorial](https://kyma-project.io/docs/1.12/components/knative-eventing-mesh/#tutorials-configure-the-kafka-channel).

## Compass

### Provisioning of custom components 

With the recent addition to the Kyma Installer, you can define the source URL of a component that you want to install. The Runtime Provisioner now fully supports this feature, allowing you to include external components that you want to deploy on provisioned clusters. 

### On-demand revision of the Kyma version on provisioned clusters 

From now on, you can specify a version of the Kyma Installer to be deployed on a provisioned cluster. This can help you to debug, track, or fix issues both during development and in the production-ready environment.

### Upgrades in the Runtime Provisioner 

> Starting with this release, the Runtime Provisioner allows you to upgrade your Kyma runtime using the [GraphQL API](https://github.com/kyma-incubator/compass/blob/master/docs/internal/runtime-upgrade.md). Though simple, the API allows you to upgrade the existing Runtime to the new Kyma version and replace the existing configuration with the one passed with the upgrade mutation. We also included a rollback option, but it may still require additional actions on the cluster. 

Reviewer 2: ```suggestion
Starting with this release, the Runtime Provisioner allows you to upgrade your Kyma Runtime to the newer version using the [GraphQL API](https://github.com/kyma-incubator/compass/blob/master/docs/internal/runtime-upgrade.md). The new configuration passed with the upgrade mutation replaces the existing one. We also included a rollback option. Bear in mind that the rollback operation may still require additional actions performed on the cluster. 
```

- New **Functions** view that allows you to easily manage Triggers, ServiceBindings, and environment variables for Functions. 

![Functions view](./functions-view.png) 

- New **Diff** option with the preview of changes in the Function's code. 

![Diff preview](./diff.png) 

- New **Resources and replicas** section in which you can manage requests and limits for the Function's resources. 

![Resources and replicas](./resources-and-replicas.png) 

### Documentation 

There is a brand-new set of docs for the new solution with detailed architecture, Function CR description and configuration options, and step-by-step tutorials for both CLI and UI: 

- [Create a Function](https://kyma-project.io/docs/1.12/components/serverless/#tutorials-create-a-function) 
- [Expose a Function with an API Rule](https://kyma-project.io/docs/1.12/components/serverless/#tutorials-expose-a-function-with-an-api-rule) 
- [Bind a Service Instance to a Function](https://kyma-project.io/docs/1.12/components/serverless/#tutorials-bind-a-service-instance-to-a-function) 
- [Trigger a Function with an event](https://kyma-project.io/docs/1.12/components/serverless/#tutorials-trigger-a-function-with-an-event) 

> > **CAUTION:** If you already use Kubeless Functions, you must migrate them manually. Read the [migration guide](https://github.com/kyma-project/kyma/blob/release-1.12/docs/migration-guides/1.11-1.12.md#serverless) for details.

Reviewer: ```suggestion
> **CAUTION:** If you already use Kubeless Functions, you must migrate them manually after you upgrade to the new release. Read the [migration guide](https://github.com/kyma-project/kyma/blob/release-1.12/docs/migration-guides/1.11-1.12.md#serverless) for details.
```

### Documentation updates

With Event Bus removed, we focused on Knative Eventing Mesh. For 1.12, this also meant providing you with additional documentation on [event processing and delivery](https://kyma-project.io/docs/1.12/components/knative-eventing-mesh/#details-event-processing-and-delivery). If you are interested in setting Kafka as your default Channel implementation, read this [tutorial](https://kyma-project.io/docs/1.12/components/knative-eventing-mesh/#tutorials-configure-the-kafka-channel).

## Compass

### Provisioning of custom components 

With the recent addition to the Kyma Installer, you can define the source URL of a component that you want to install. The Runtime Provisioner now fully supports this feature, allowing you to include external components that you want to deploy on provisioned clusters. 

### On-demand revision of the Kyma version on provisioned clusters 

From now on, you can specify a version of the Kyma Installer to be deployed on a provisioned cluster. This can help you to debug, track, or fix issues both during development and in the production-ready environment.

### Upgrades in the Runtime Provisioner 

> In this release, we introduced the first version of [GraphQL API](https://github.com/kyma-incubator/compass/blob/master/docs/internal/runtime-upgrade.md) the Runtime Provisioner uses to upgrade a Kyma Runtime. Though simple, the API allows you to upgrade the existing Runtime to the new Kyma version and replace the existing configuration with the one passed with the upgrade mutation. We also included a rollback option, but it may still require additional actions on the cluster. 

Reviewer 2: The Runtime Provisioner has had a GraphQL API for a while now, it's just the Upgrades addition that's introduced in this release, that is, the `upgradeRuntime` and `rollBackUpgradeOperation` mutations. To learn more, read [this](https://github.com/kyma-incubator/compass/blob/master/docs/internal/runtime-upgrade.md) document (please don't link to this document, as this is just for developers).

Author: I used the doc to rephrase this line :)

### Automatic status update for Applications and Runtimes in the Director 

> When a paired Application or Runtime communicates with the Director API for the first time, the Director automatically sets its status to **CONNECTED**. This operation does not prevent you or the Integration System from managing Application and Runtime status on your own.

Reviewer 2: ```suggestion
When a paired Application or Runtime communicates with the Director API for the first time, the Director automatically sets its status to **CONNECTED**. This operation does not prevent you or the Integration System from managing Application and Runtime statuses on your own.
```
Fixing it after my previous suggestion. Sorry for the confusion 😝

> When a paired Application or Runtime communicates with the Director API for the first time, the Director automatically sets its status to **CONNECTED**. This operation does not prevent you or the Integration System from managing Application and Runtime statuses on your own.

Reviewer 2: ```suggestion
When a paired Application or Runtime communicates with the Director API for the first time, the Director automatically sets its status to **CONNECTED**. This operation does not prevent you or the Integration System from managing Application and Runtime status on your own.
```
I'm not 100% sure if the `status` here scopes just to the Runtime or to the Application, too. @klaudiagrz , can you confirm? (The reason I'm not sure is we have a custom resource called `Application` and a mutation in Compass to get Runtime status).

Reviewer 3: The status scopes to both Runtime and Application.

Reviewer 2: I see. Thanks, Klaudia!
Ups, @bszwarc, my suggestion is wrong then - too fast and too furious :D 
I'll add a new one to fix it.

### Automatic Scenario Assignment in the Director 

The [Automatic Scenario Assignment (ASA)](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-03-automatic-scenario-assignment.md) feature allows you to define a condition that specifies when a Scenario is automatically assigned to a Runtime. For example, you can specify a label that adds a given Scenario to each Runtime created by a given user, company, or any other entity. 

## Console 

### Improved browsing of APIs within packages 

In the 1.11 release, we introduced API packages to allow the Applications to register APIs grouped in packages. This way, you could instantiate entire packages and access multiple APIs at once. We now simplified browsing APIs within a package even further in the Service Catalog UI. If a package holds multiple APIs, you can find them quickly using a dedicated drop-down list with a search feature. 

![api-packages](./api-packages.png)

## Observability

### Brand new Kiali 

> For 1.12, we took the chance and upgraded the Kiali component to leverage the newest Kiali Operator, making it possible for you to configure all options in a convenient way using the Installer overrides. Although we had to rewrite the component implementation, these changes do not affect you in any way as the user.

Reviewer 2: ```suggestion
For 1.12, we took the chance and upgraded the Kiali component to leverage the newest Kiali Operator, making it possible for you to configure all options in a convenient way using the Installer overrides. Although we had to rewrite the component implementation, these changes do not affect you as the user in any way.
```

> For 1.12, we took the chance and upgraded the Kiali component to leverage the newest Kiali Operator, making it possible for you to configure all options in a convenient way using the Installer overrides. Although we had to rewrite the component implementation, these changes do not affect you as the user in any way .

Reviewer 2: ```suggestion
For 1.12, we took the chance and upgraded the Kiali component to leverage the newest Kiali Operator, making it possible for you to configure all options in a convenient way using the Installer overrides. Although we had to rewrite the component implementation, these changes do not affect you as the user in any way.
```
\*A wild space appears\*

Author: 👀 indeed, it does :O

### Jaeger renamed to Tracing 

> The Jaeger component also received a full makeover and a new name. The new **Tracing** component is based on the most recent Jaeger Operator and brings in new features and fixes. As the BadgerDB persistence was causing more problems then satisfaction due to issues with data retention and memory consumption higher than expected, we switched back to the in-memory-based deployment model. Additionally, we ensured that you can use the Installer overrides to configure deployment strategies available in Jaeger Operator. All this allows you to switch back to a BadgerDB-based deployment or even introduce a production scenario based on Elastic Search. 

Reviewer 2: ```suggestion
The Jaeger component also received a full makeover and a new name. The new **Tracing** component is based on the most recent Jaeger Operator and brings in new features and fixes. As the BadgerDB persistence caused more problems then satisfaction due to issues with data retention and memory consumption higher than expected, we switched back to the in-memory-based deployment model. Additionally, we ensured that you can use the Installer overrides to configure deployment strategies available in the Jaeger Operator. All this allows you to switch back to a BadgerDB-based deployment or even introduce a production scenario based on Elastic Search. 
```
`was causing` -> `caused`
`Jaeger Operator` -> `the Jaeger Operator`

## Monitoring

### Monitoring upgrade

We updated the monitoring stack to the latest versions of Prometheus and Grafana. By doing that we ensured that all configured monitoring targets are healthy and working, and all Grafana dashboards are consistently tagged and named. 

## Serverless

> As mentioned in the last release notes, we switched to a new version of Serverless that relies on [Knative Serving](https://knative.dev/docs/serving/) for deploying and managing Functions, and [Kubernetes Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) for creating Docker images. For this reason, backend, frontend, and documentation of Serverless have received full makeovers. 

Reviewer 2: ```suggestion
As mentioned in the previous release notes, we switched to a new version of Serverless that relies on [Knative Serving](https://knative.dev/docs/serving/) for deploying and managing Functions, and [Kubernetes Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) for creating Docker images. For this reason, backend, frontend, and documentation of Serverless have received full makeovers. 
```
I read somewhere that `last` means `ultimate`, `final`, and is misused in the sense of `previous`. But there are different opinions in different sources, so that's more of an open question than a suggestion.

### Backend

> Serverless is now based on a new [Function (CR)](https://kyma-project.io/docs/1.12/components/serverless/#custom-resource-function) that provides you with full control over resource management, replacing at the same time the previous Function sizes management. We also switched to the Node.js 12 Runtime and improved the [resource processing flow](https://kyma-project.io/docs/1.12/components/serverless/#details-function-processing) in which the Function's image is built in a separate phase, helping the Functions to run more smoothly. 

Reviewer 2: ```suggestion
Serverless is now based on the new [Function CR](https://kyma-project.io/docs/1.12/components/serverless/#custom-resource-function) that provides you with full control over resource management, replacing at the same time the previous Function sizes management. We also switched to the Node.js 12 Runtime and improved the [resource processing flow](https://kyma-project.io/docs/1.12/components/serverless/#details-function-processing) in which the Function's image is built in a separate phase, helping the Functions to run more smoothly. 
```
`a new Function (CR)` -> `the new Function CR`

I'm not dead keen on the parentheses. Do we need them?

Reviewer: yes, of course, without parentheses, that's an oversight - @majakurcius thanks for spotting that :)

Author: Yep, should be removed when I removed `custom resource` - my bad. :)

> Serverless is now based on a new [Function custom resource (CR)](https://kyma-project.io/docs/1.12/components/serverless/#custom-resource-function) that provides you with full control over resource management, replacing at the same time the previous Function sizes management. We also switched to the Node.js 12 Runtime and improved the [resource processing flow](https://kyma-project.io/docs/1.12/components/serverless/#details-function-processing) in which the Function's image is built in a separate phase, helping the Functions to run more smoothly. 

Reviewer: new [Function custom resource (CR)] -> new [Function CR]

### On-demand revision of the Kyma version on provisioned clusters 

From now on, you can specify a version of the Kyma Installer to be deployed on a provisioned cluster. This can help you to debug, track, or fix issues both during development and in an production-ready environment.

### Upgrades in Compass Provisioner 

In this release, we introduced the first version of [GraphQL API](https://github.com/kyma-incubator/compass/blob/master/docs/internal/runtime-upgrade.md) the Runtime Provisioner uses to upgrade a Kyma Runtime. Though simple, the API allows you to upgrade an existing Runtime to the new Kyma version and replace the existing configuration with the one passed with the upgrade mutation. We also included a rollback option, but it may still require additional actions on the cluster. 

### Automatic status update for Applications and Runtimes in Compass Director 

When a paired Application or Runtime communicates with the Director API for the first time, the Director automatically sets its status to **CONNECTED**. This operation does not prevent you or the integration system from managing Application and Runtime statuses on your own.

> ### Automatic Scenario Assignment in Compass Director 

Author: the Director?

The [Automatic Scenario Assignment (ASA)](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-03-automatic-scenario-assignment.md) feature allows you to define a condition that specifies when a Scenario is automatically assigned to a Runtime. For example, you can specify a label that adds a given Scenario to each Runtime created by a given user, company, or any other entity. 

## Console 

### Improved browsing of APIs within packages 

> In the 1.11 release we introduced API packages to allow the Applications to register APIs grouped in packages. This way, you could instantiate entire packages and access multiple APIs at once. We now simplified browsing APIs within a package even further in the Service Catalog UI. If a package holds multiple APIs, you can find them quickly using a dedicated drop-down component with a search feature. 

Reviewer: ![image](https://user-images.githubusercontent.com/40655785/80599775-c6b1f680-8a2b-11ea-821d-2ba6935d64b8.png)

> In the 1.11 release, we introduced API packages to allow the Applications to register APIs grouped in packages. This way, you could instantiate entire packages and access multiple APIs at once. We now simplified browsing APIs within a package even further in the Service Catalog UI. If a package holds multiple APIs, you can find them quickly using a dedicated drop-down component with a search feature. 

Author: List sounds good :)

![api-packages](./api-packages.png)

## Observability

### Brand new Kiali 

For 1.12, we took the chance and upgraded the Kiali component to leverage the newest Kiali Operator, making it possible for you to configure all options in a convenient way using the Installer overrides. Although we had to rewrite the component implementation, these changes do not affect you in any way as the user.

### Jaeger renamed to Tracing 

The Jaeger component also received a full makeover and a new name. The new, **Tracing** component is based on the most recent Jaeger Operator and brings in new features and fixes. As the BadgerDB persistence was causing more problems then satisfaction due to issues with data retention and memory consumption higher than expected, we switched back to the in-memory-based deployment model. Additionally, we ensured that you can use the Installer overrides to configure deployment strategies available in Jaeger Operator. All this allows you to switch back to a BadgerDB-based deployment or even introduce a production scenario based on Elastic Search. 

> 

Reviewer: I think I would  remove the comma from `The new, **Tracing** component`

Author: Removed :)

## Monitoring

### Monitoring upgrade

We updated the monitoring stack to the latest versions of Prometheus and Grafana. By doing that we ensured that all configured monitoring targets are healthy and working, and all Grafana dashboards are consistently tagged and named. 

## Serverless

> As mentioned in the last release notes, we switched to a new version of Serverless that relies on [Knative Serving](https://knative.dev/docs/serving/) for deploying and managing Functions and [Kubernetes Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) for creating Docker images. For this reason, backend, frontend, and documentation of Serverless have received full makeovers. 

Author: I wouldn't add a comma there, just leaving at as Functions and Kubernetes Jobs 🤔

### Backend

> Serverless is now based on a new [Function custom resource (CR)](https://kyma-project.io/docs/1.12/components/serverless/#custom-resource-function) that provides you with full control over resource management (including scale-to-zero), replacing at the same time the previous Function sizes management. We also switched to the Node.js 12 Runtime and improved the [resource processing flow](https://kyma-project.io/docs/1.12/components/serverless/#details-function-processing) in which the Function's image is built in a separate phase, helping the Functions to run more smoothly. 

Reviewer: ```suggestion
Serverless is now based on a new [Function custom resource (CR)](https://kyma-project.io/docs/1.12/components/serverless/#custom-resource-function) that provides you with full control over resource management, replacing at the same time the previous Function sizes management. We also switched to the Node.js 12 Runtime and improved the [resource processing flow](https://kyma-project.io/docs/1.12/components/serverless/#details-function-processing) in which the Function's image is built in a separate phase, helping the Functions to run more smoothly. 
```

Reviewer: 1. I removed scale-to-zero ;)...as it turned out to be invalid
2. Pls just write Function CR instead of `[Function custom resource (CR)]` as CR is already introduced above in the doc

### Frontend

Changes in the Console UI include: 

- New **Functions** view that allows you to easily manage Triggers, ServiceBindings, and environment variables for Functions. 

![Functions view](./functions-view.png) 

- New **Diff** option with the preview of changes in the Function's code. 

![Diff preview](./diff.png) 

> - New **Resources and replicas** section in which you can manage requests and limits for the Function's resources and scale the Function to zero. 

Reviewer: ```suggestion
- New **Resources and replicas** section in which you can manage requests and limits for the Function's resources. 
```

> ![Resources and replicas](./resources-and-replicas.png) 

Author: This will be updated.

> ![Resources and replicas](./resources-and-replicas.png) 

Reviewer: This screenshot will no longer be valid so I will provide you with a new one shortly ;)

Reviewer: ![resources-and-replicas](https://user-images.githubusercontent.com/40655785/80604704-5a86c100-8a32-11ea-8bc9-7e759f7e442f.png)

Author: Replaced :)

>**NOTE:** If the migration guide includes any additional instructions, make sure to apply them manually. 

### Automatic status update for Applications and Runtimes in Compass Director 

When a paired Application or Runtime communicates with the Director API for the first time, the Director automatically sets their status to **CONNECTED**. This operation does not prevent you or the Integration System from managing Application and Runtime status on your own.

### Automatic Scenario Assignment in Compass Director 

The [Automatic Scenario Assignment (ASA)](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-03-automatic-scenario-assignment.md) feature allows you to define a condition that specifies when a Scenario is automatically assigned to a Runtime. For example, you can specify a label that adds a given Scenario to each Runtime created by a given user, company, or any other entity. 

## Console 

### Improved browsing of APIs within packages 

> In 1.11 release we introduced API packages to allow the Applications to register of APIs grouped in packages. This way, you could instantiate entire packages and access multiple APIs at once. Now, with 1.12, browsing APIs within package in Service Catalog UI is even simpler. In case a package holds multiple APIs, you can find them quickly using a dedicated drop-down component with a search feature. 

Reviewer 4: ```suggestion
In 1.11 release we introduced API packages to allow the Applications to register APIs grouped in packages. This way, you could instantiate entire packages and access multiple APIs at once. Now, with 1.12, browsing APIs within package in Service Catalog UI is even simpler. In case a package holds multiple APIs, you can find them quickly using a dedicated drop-down component with a search feature. 
```

> In 1.11 release we introduced API packages to allow the Applications to register of APIs grouped in packages. This way, you could instantiate entire packages and access multiple APIs at once. Now, with 1.12, browsing APIs within package in Service Catalog UI is even simpler. In case a package holds multiple APIs, you can find them quickly using a dedicated drop-down component with a search feature. 

Reviewer: ```suggestion
In the 1.11 release, we introduced API packages to allow the Applications to register APIs grouped in packages. This way, you could instantiate entire packages and access multiple APIs at once. We now made browsing APIs within a package even simpler in the Service Catalog UI. If a package holds multiple APIs, you can find them quickly using a dedicated drop-down component with a search feature. 
```

![api-packages](./api-packages.png)

## Observability

### Brand new Kiali 

> For 1.12, we took the chance and upgraded the Kiali component to leverage the newest Kiali Operator, making it possible for you to configure all options in a convenient way using the Installer overrides. Although  we had to rewrite the component implementation, these changes do not affect you in any way as the user.

Reviewer: ```suggestion
For 1.12, we took the chance and upgraded the Kiali component to leverage the newest Kiali Operator, making it possible for you to configure all options in a convenient way using the Installer overrides. Although we had to rewrite the component implementation, these changes do not affect you in any way as the user.
```

### Jaeger renamed to Tracing 

> The Jaeger component also received a full makeover and a new name. The new,  **Tracing** component is based on the most recent Jaeger Operator and brings in new features and fixes. As the BadgerDB persistence was causing more problems then satisfaction due to issues with data retention and memory consumption higher than expected, we switched back to the in-memory-based deployment model. Additionally, we ensured that you can use the Installer overrides to configure deployment strategies available in Jaeger Operator. All this allows you to switch back to a BadgerDB-based deployment or even introduce a production scenario based on Elastic Search. 

Reviewer: ```suggestion
The Jaeger component also received a full makeover and a new name. The new, **Tracing** component is based on the most recent Jaeger Operator and brings in new features and fixes. As the BadgerDB persistence was causing more problems then satisfaction due to issues with data retention and memory consumption higher than expected, we switched back to the in-memory-based deployment model. Additionally, we ensured that you can use the Installer overrides to configure deployment strategies available in Jaeger Operator. All this allows you to switch back to a BadgerDB-based deployment or even introduce a production scenario based on Elastic Search. 
```

Reviewer: Meh, GitHub doesn't show that but I guess it was only an additional space here `The new,  **Tracing** `

## Monitoring

### Monitoring upgrade

We updated the monitoring stack to the latest versions of Prometheus and Grafana. By doing that we ensured that all configured monitoring targets are healthy and working, and all Grafana dashboards are consistently tagged and named. 

## Serverless

As mentioned in the last release notes, we switched to a new version of Serverless that relies on [Knative Serving](https://knative.dev/docs/serving/) for deploying and managing Functions and [Kubernetes Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) for creating Docker images. For this reason, backend, frontend, and documentation of Serverless have received full makeovers. 

### Serverless based on Function CR 

> 

Reviewer: There were more changes in the backend than just introducing a new CR that is why I would prefer the generic `Backed` title I originally proposed - same for the frontend.

Author: I changed it :D

> Serverless is now based on a new [Function custom resource (CR)](https://kyma-project.io/docs/1.12/components/serverless/#custom-resource-function) that provides you with full control over resource management (including scale-to-zero), replacing the previous Function sizes management. We also switched to the Node.js 12 Runtime and improved the [resource processing flow](https://kyma-project.io/docs/1.12/components/serverless/#details-function-processing) in which the Function's image is built in a separate phase, helping the Functions to run more smoothly. 

Reviewer: ```suggestion
Serverless is now based on a new [Function CR](https://kyma-project.io/docs/1.12/components/serverless/#custom-resource-function) that provides you with full control over resource management (including scale-to-zero), replacing at the same time the previous Function sizes management. We also switched to the Node.js 12 Runtime and improved the [resource processing flow](https://kyma-project.io/docs/1.12/components/serverless/#details-function-processing) in which the Function's image is built in a separate phase, helping the Functions to run more smoothly. 
```
